cli/flox/src/commands/gc.rs

```diff
@@ -1,23 +1,291 @@
+
+    fn is_deletion_line(line: &str) -> bool {
+        line.starts_with("deleting '/nix/store")
+    }
+
+    fn is_done_line(line: &str) -> bool {
+        line.ends_with("MiB Freed")
+    }
+
+    fn new_progress_state(&self, line: &str) -> Option<GcProgress> {
+        use GcProgress::*;
+        match self {
+            Init => {
+                if GcProgress::is_finding_roots_line(line) {
+                    Some(GcProgress::FindingRoots)
+                } else {
+                    None
+                }
+            },
+            FindingRoots => {
+                if GcProgress::is_removing_links_line(line) {
+                    Some(GcProgress::RemovingLinks)
+                } else if GcProgress::is_scanning_line(line) {
+                    Some(GcProgress::Scanning)
+                } else if GcProgress::is_deletion_line(line) {
+                    //  There may not be stale links, so we should detect
+                    //  the 'deleting...' line as well and possibly skip
+                    //  this stage if necessary
+                    let msg = Self::capitalize_deleting_store_path_line(line);
+                    Some(GcProgress::DeletingStorePaths(msg))
+                } else {
+                    None
+                }
+            },
+            RemovingLinks => {
+                if GcProgress::is_deletion_line(line) {
+                    // Need to capitalize the 'd' so it matches the case of
+                    // the other log messages.
+                    let msg = Self::capitalize_deleting_store_path_line(line);
+                    Some(GcProgress::DeletingStorePaths(msg))
+                } else if GcProgress::is_scanning_line(line) {
+                    Some(GcProgress::Scanning)
+                } else {
+                    None
+                }
+            },
+            Scanning => {
+                if GcProgress::is_deletion_line(line) {
+                    // Need to capitalize the 'd' so it matches the case of
+                    // the other log messages.
+                    let msg = Self::capitalize_deleting_store_path_line(line);
+                    Some(GcProgress::DeletingStorePaths(msg))
+                } else {
+                    None
+                }
+            },
+            DeletingStorePaths(_msg) => {
+                if GcProgress::is_deletion_line(line) {
+                    let msg = Self::capitalize_deleting_store_path_line(line);
+                    Some(GcProgress::DeletingStorePaths(msg))
+                } else if GcProgress::is_done_line(line) {
+                    Some(Done(line.to_string()))
+                } else {
+                    None
+                }
+            },
+            Done(_msg) => None,
+        }
+    }
+
+    fn capitalize_deleting_store_path_line(line: &str) -> String {
+        let (_d, rest) = line.split_at(1);
+        let mut msg = String::new();
+        msg.push('D');
+        msg.push_str(rest);
+        msg
```

Comment:
so essentially this? 

```suggestion
        line.to_string().replace_range(0..=1, "D")
```